### ISSUE TITLE
Relicense under an OSI approved license

### ISSUE THREAD

**Issue by JamieMagee** [Genie-Garage/aladdin-python-sdk#2]
The current license of this package is a proprietary license[^1]. As a result, it is not compatible with the Open Source license, Apache 2.0, used by Home Assistant[^2] and the Aladdin Connection integration is set to be removed from Home Assistant 2024.7.0[^3].

You previously stated that you would have to re-start a discussion with your legal team about releasing the library under an OSI approved license[^4]. As a software developer that works in a large enterprise, I can empathise. However, you already signed the Contributor License Agreement[^5] for the the code you contributed to Home Assistant Core, which released your contributions under the Apache 2.0 License[^6]. It shouldn't be too much of a stretch to release this library under the Apache 2.0 license as well. Nothing in this library is novel.

You previously committed to creating and maintaining an official Home Assistant integration when you sunset the old API[^7]. I assume you were aware of the requirements of contributing to Home Assistant when you made that commitment. So I am asking you to honor it.

[^1]: https://github.com/Genie-Garage/aladdin-python-sdk/blob/main/LICENSE
[^2]: https://github.com/home-assistant/core/blob/dev/LICENSE.md
[^3]: https://github.com/home-assistant/core/pull/120980
[^4]: https://github.com/home-assistant/core/pull/120029#issuecomment-2181081071
[^5]: https://github.com/home-assistant/core/pull/117020#event-12732266518
[^6]: https://www.home-assistant.io/developers/cla/
[^7]: https://alerts.home-assistant.io/alerts/aladdin_connect_2/

**Comment by shawngmc** [comment]
Has any internal discussion occurred on this? This is a substantial blocker for home automation integrations.

Another option might be to make the API interface documentation public, at which point an Open Source-compliant implementation could be made with a reasonable expectation that the docs would be kept up to date.

**Comment by norberttech** [comment]
Any news on this? Its a shame that due to lack of understanding of how irrelevant and simple those 3 files are, home assistant integration had to be removed. Legal team that have "concerns" in this case should be fired, as they are working against the best interest of the company just saying "no" to whatever they don't understand...

**Comment by norberttech** [comment]
yeah, we bought one just a month or two before it was removed from HA and that was THE ONLY reason to go with Genie. All because group of useless lawyers who only cares about charging more doesn't understand that there is NOTHING proprietary in this so called "sdk". Its a code that I can be written during morning coffee (if you drink espresso) m

**Comment by cbenard** [comment]
+1 for needing this

**Comment by amar5150** [comment]
+1 this as well. The api in HA worked flawlessly for me and was used very often by my family. I have recommended Aladdin openers to others for this reason alone so its a shame that this feature was removed due to some legal wording.

**Comment by cbenard** [comment]
> Any news on this? Its a shame that due to lack of understanding of how irrelevant and simple those 3 files are

Looking at it after your comment, I completely agree. Really hard to believe.

Do you know Python 3? I could fumble my way through it, but I'm wondering if I documented:

* The very simple REST API
* Request/response models
* Authentication (just bearer auth header + api key header)
* The SDK API (classes, models, functions)

Could you then write an API client for it that corresponds to those in a "clean room" way? I don't have any confidence their lawyers are going to be able to pull their heads from where they currently reside.

My other alternative is just to clone the code from the past commit and just use it, because I don't care if it's OSI-compliant for my own use, but I would hate to have normies locked out of it forever.

**Comment by billvinson** [comment]
Any updates would be appreciated. I bought 2 Aladdin Genie garage door openers purely based on the fact that they we interoperable with things like Home Assistant. Then it got removed with a promise to reissue it as an official intgration. I had assumed you all had just failed to keep that commitment as I had missed 2024.6 (reintroduction of the integration) & 2024.7 (removal of the integration due to non-compliant license).

Right now, my opinion of ODC is low and I'm considering hardwiring a solution to ignore the "smart" part of the devices completely. I'm hoping you all can do the right thing soon and fix this. I'd like to go back to being able to recommend these devices to others, but that is completely impossible at the moment.

**Comment by CoolSilver** [comment]
Support had to go around a bit to finally say there was this license concern. Sadly Genie caused the problem, yet other closed platforms get the benefit.

"Home assistant integration is being held up due to legal concerns with the end user license agreement. No word on when it will be resolved. "

**Comment by hugh-martin** [comment]
+1. I will be able to vote in the next couple of months with my wallet. We have Genie in our current home and have been very happy with them, but we’re building a new home and at this point, we won’t be putting in new Genie garage door openers.

**Comment by brandond** [comment]
Commenting here to +1 on this being a dealbreaker for me. Will definitely not be recommending against Genie as long as they have such inane licensing on open-source API bindings.

**Comment by ez12a** [comment]
For those that are tired of waiting and to avoid potential upstream issues like this altogether I just got a Meross wifi controller off of Amazon. Makes any dumb opener smart and it works on the Genie Model 3120. Installation was not hard. Glad to have control of the genie back in Home Assistant again.

**Comment by amar5150** [comment]
If you don't mind having some reliance on the cloud, adding the genie into
Smart things or Google Home, then integrating one of those into Home
Assistant also works. That's what I'm doing until this is situated.

**Comment by justinjahns** [comment]
Another excellent alternative is Konnected... Works great with my Genie 7155 and should work with any Genie model. It's an ESPHome device, so easily integrates with HA and all local control. This was my first experience with ESPHome, so I used the Konnected app to build the ESPHome config and apply it to the device - they made it very easy!

**Comment by yanismydj** [comment]
This issue has earned Genie a spot on the Internet of Trash hall of shame https://github.com/unixorn/internet-of-trash/pull/17

**Comment by JamieMagee** [comment]
I emailed asking if there were any updates and I received this response:

> Unfortunately, I do not have any updates or timetable on this. The integration is being help up due to legal concerns with the end user license agreement. No word on when it will be resolved. Thank you for your patience.

**Comment by JamieMagee** [comment]
If Genie is unable or unwilling to release this library under an open source license that is compatible Apache 2.0, [a clean-room design][1] would likely be the only path forward:

> Typically, a clean-room design is done by having someone examine the system to be reimplemented and having this person write a specification. This specification is then reviewed by a lawyer to ensure that no copyrighted material is included. The specification is then implemented by a team with no connection to the original examiners.

Unfortunately, that would mean that anyone who has commented or interacted with this issue would not be able to work on an implementation.

[1]: https://en.wikipedia.org/wiki/Clean-room_design

**Comment by brandond** [comment]
This repo has seen no changes since shortly after it was released. Is Genie really just going to leave an abandonware SDK out there with an unusable license? This is unacceptable.

**Comment by erikreedstrom** [comment]
HACS repackaging of the original core component: [Aladdin Connect](https://github.com/erikreedstrom/aladdin_connect)

If you engage in personal use in the US or Canada, there shouldn't be any conflict with the proprietary license.

Much appreciation to Genie for making this SDK available. 🙇

**Comment by brandond** [comment]
@frenck @JamieMagee can you reopen https://github.com/home-assistant/core/issues/119697? Unfortunately y'all locked the HA issue so we can't continue the discussion on that side now that this issue has been resolved.

**Comment by brandond** [comment]
lol this is the [MIT license](https://opensource.org/license/mit) with `or personal use only` awkwardly added to the first sentence. The paragraph wasn't re-wrapped properly, and somehow the first word on the next line got capitalized despite it being part of the same sentence.

**Comment by edmonds** [comment]
> So what if I were to (for personal use only of course) modify and sublicense this project under a proper MIT license? That's allowed by the terms of THIS weird license, and HA and other projects could then use it without encumbrance.

Personally I wouldn't mess with someone who can let themselves into my garage.

**Comment by swcloudgenie** [comment]
We have updated to use the MIT license as is

**Comment by JamieMagee** [comment]
I opened pull requests to re-add the integration:
- https://github.com/home-assistant/core/pull/149029
- https://github.com/home-assistant/home-assistant.io/pull/40068
- https://github.com/home-assistant/brands/pull/7485

**Comment by patienttruth** [comment]
> We have updated to use the MIT license as is

Thank you!
